embedded/EscapeROOM/radio_morse_panel/radio_morse_panel.c

```diff
@@ -22,139 +48,84 @@ void unit()
 	}
 }
 
-void dot()
+void PlayDot()
 {
 	carrier = CARRIER_ON;
-	unit();
+	PlayUnit();
 	carrier = CARRIER_OFF;
 }
 
-void dash()
+void PlayDash()
 {
 	carrier = CARRIER_ON;
-	unit(); unit(); unit();
+	PlayUnit(); PlayUnit(); PlayUnit();
 	carrier = CARRIER_OFF;
 }
 
-void short_space()
+void PlayShortSpace()
 {
-	unit();
+	PlayUnit();
 }
 
-void letter_space()
+void PlayLetterSpace()
 {
-	unit(); unit(); unit();
+	PlayUnit(); PlayUnit(); PlayUnit();
 }
 
-void word_space()
+void PlayWordSpace()
 {
-	unit(); unit(); unit(); unit(); unit(); unit(); unit();
+	PlayUnit(); PlayUnit(); PlayUnit(); PlayUnit(); PlayUnit(); PlayUnit(); PlayUnit();
 }
 
-void zero()
-{
-	dash(); short_space();
-	dash(); short_space();
-	dash(); short_space();
-	dash(); short_space();
-	dash();
-}
-
-
-void one()
-{
-	dot();  short_space();
-	dash(); short_space();
-	dash(); short_space();
-	dash(); short_space();
-	dash();
-}
-
-
-void two()
-{
-	dot(); short_space();
-	dot(); short_space();
-	dash(); short_space();
-	dash(); short_space();
-	dash();
-}
-
-
-void three()
-{
-	dot(); short_space();
-	dot(); short_space();
-	dot(); short_space();
-	dash(); short_space();
-	dash();
-}
-
-
-void four()
-{
-	dot(); short_space();
-	dot(); short_space();
-	dot(); short_space();
-	dot(); short_space();
-	dash();
-}
-
-
-void five()
-{
-	dot(); short_space();
-	dot(); short_space();
-	dot(); short_space();
-	dot(); short_space();
-	dot();
-}
-
-void six()
-{
-	dash(); short_space();
-	dot(); short_space();
-	dot(); short_space();
-	dot(); short_space();
-	dot();
-}
-
-void seven()
-{
-	dash(); short_space();
-	dash(); short_space();
-	dot(); short_space();
-	dot(); short_space();
-	dot();
+unsigned char GetLetterPositionInMorseAlphabet(char Letter, const MorseCode *MorseAlphabet, const unsigned char NumOfElements){
+	unsigned char LetterPosition;
+	
+	// simply compare each element in alphabet;
+	for(LetterPosition = 0; LetterPosition < NumOfElements; LetterPosition++){
+		if( (MorseAlphabet + LetterPosition)->Symbol == Letter)
+			break;
+	}
+	
+	return LetterPosition;
 }
 
-void eight()
-{
-	dash(); short_space();
-	dash(); short_space();
-	dash(); short_space();
-	dot(); short_space();
-	dot();
+void PlayMorseLetter(char Letter, const MorseCode *MorseAlphabet, const unsigned char NumOfElements){
+	unsigned char LetterPositionInAlphabet;
+	unsigned char i;
+	
+	LetterPositionInAlphabet = GetLetterPositionInMorseAlphabet(Letter, MorseAlphabet, NumOfElements);
+	
+	// play each morse letter code element 
+	for(i = 0; i < MAX_MORSE_PARTS && (MorseAlphabet + LetterPositionInAlphabet)->Code[i] != '\0'; i++){
```

Comment:
Why you use (name + index)->element notation? Is it somehow easier to understand than name[index].element?